cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/SlangCompiler.java

```diff
@@ -27,10 +28,19 @@
     /**
      * Pre-compile a CloudSlang source into an {@link io.cloudslang.lang.compiler.modeller.model.Executable}
      * @param source the {@link SlangSource}
-     * @return an {@link io.cloudslang.lang.compiler.modeller.model.Executable} object, containing either a flow or operation in the file.
+     * @return an {@link io.cloudslang.lang.compiler.modeller.model.Executable} object, containing either a flow or
+     * operation in the file.
      */
     Executable preCompile(SlangSource source);
 
+    /**
+     * Pre-compile a CloudSlang source into an {@link ExecutableModellingResult}
```

Comment:
I think we should mention in the javadoc that this version will collect the exceptions instead of failing